src/Control/HTTP.php

```diff
@@ -268,7 +268,7 @@ public static function setGetVar($varname, $varvalue, $currentURL = null, $separ
         $newUri = $scheme . '://' . $user . $host . $port . $path . $params . $fragment;
 
         if ($isRelative) {
-            return Director::makeRelative($newUri);
+            return Director::baseURL() . Director::makeRelative($newUri);
```

Comment:
By definition this is no longer returning a relative link. I have a conceptual issue with that change

Reply:
> By definition this is no longer returning a relative link. I have a conceptual issue with that change

There is no mention of this function returning a relative link, in fact the docblock clearly states:
```
* CAUTION: If the URL is determined to be relative, it is prepended with Director::absoluteBaseURL().
* This method will always return an absolute URL because Director::makeRelative() can lead to
* inconsistent results.
```
The thing is it doesn't "always return an absolute URL".

Anyway, this was an attempt at the least invasive approach to fix the issue. It sounds to me like you are proposing a different core change, so I'm not sure what to do next (or even if I should). This is I think best left to you guys.

Reply:
That's fair enough =) I'll remove my concern on that issue